src/playlist-loader.js

```diff
@@ -255,11 +294,48 @@ export default class PlaylistLoader extends EventTarget {
         // only refresh the media playlist if no other activity is going on
         return;
       }
+      const media = this.media();
+
+      let uri = resolveUrl(this.master.uri, media.uri);
+
+      if (this.experimentalLLHLS) {
```

Comment:
This is all new, basically use query directives and server side blocking if available.

Reply:
question: do we want to skip these two server-control checks if we have an end list tag? The spec says that the server MUST ignore these, so, we may as well not do the calculations in those cases, if we don't already.

Reply:
sure I will make this an `&& !endList`